eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/InjectEntityManagerMappingsPass.php

```diff
@@ -0,0 +1,129 @@
+                }
+
+                $container->setDefinition($metadataDriverServiceName, $metadataDriverDefinition);
+
+                foreach ($driverPaths as $prefix => $driverPath) {
+                    $chainMetadataDriverDefinition->addMethodCall(
+                        'addDriver',
+                        [new Reference($metadataDriverServiceName), $prefix]
+                    );
+                }
+            }
+        }
+    }
+
+    private function createMetadataDriverDefinition($driverType, $driverPaths): Definition
+    {
+        $metadataDriver = new Definition("%doctrine.orm.metadata.{$driverType}.class%");
+        $arguments = [];
+
+        if ('annotation' === $driverType) {
+            $arguments[] = new Reference('doctrine.orm.metadata.annotation_reader');
+        }
+
+        $arguments[] = array_values($driverPaths);
+
+        $metadataDriver->setArguments($arguments);
+        $metadataDriver->setPublic(false);
+
+        return $metadataDriver;
+    }
+
+    private function prepareMappingDriverConfig(array $entityManagerConfig, ContainerBuilder $container): array
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+        $driverConfig = [];
+        foreach ($entityManagerConfig as $mappingName => $config) {
+            $config = array_replace([
+                'dir' => false,
+                'type' => false,
+                'prefix' => false,
+            ], (array) $config);
+
+            $config['dir'] = $container->getParameterBag()->resolveValue($config['dir']);
+
+            if ($config['is_bundle']) {
+                $bundle = null;
+                foreach ($bundles as $bundleName => $class) {
+                    if ($mappingName === $bundleName) {
+                        $bundle = new \ReflectionClass($class);
+
+                        break;
+                    }
+                }
+
+                if (null === $bundle) {
+                    throw new \InvalidArgumentException(sprintf('Bundle "%s" does not exist or it is not enabled.', $mappingName));
+                }
+
+                $config = $this->getMappingDriverBundleConfigDefaults($config, $bundle, $container);
+            }
+
+            if (!is_dir($config['dir'])) {
+                throw new \InvalidArgumentException(sprintf('Invalid Doctrine mapping path given. Cannot load Doctrine mapping/bundle named "%s".', $mappingName));
+            }
+
+            $driverConfig[$config['type']][$config['prefix']] = realpath($config['dir']) ?: $config['dir'];
+        }
+
+        return $driverConfig;
+    }
+
+    private function getMappingDriverBundleConfigDefaults(array $bundleConfig, \ReflectionClass $bundle, ContainerBuilder $container)
+    {
+        $bundleDir = \dirname($bundle->getFileName());
+
+        if (!$bundleConfig['type'] || !$bundleConfig['dir'] || !$bundleConfig['prefix']) {
+            return false;
```

Comment:
This looks invalid. According to it's usage above this should always return an array (or we are missing checks).